src/index.jsx

```diff
@@ -254,8 +258,10 @@ class LazyLoad extends Component {
     checkVisible(this);
   }
 
-  shouldComponentUpdate() {
-    return this.visible;
+  shouldComponentUpdate(nextProps) {
+    return this.visible &&
+      this.visible !== this.previousVisibility &&
+      !isEqual(this.props, nextProps);
```

Comment:
this is likely to hamper performance rather than improve it - its a deep equal check over all the props which should be avoided
better off using `shallowequal` - see: https://medium.com/@jooj123/react-dos-and-don-ts-1177fb12351a (point 7)

Reply:
shallowEqual might not catch all cases. but if you guys are fine for it, no problem for me as well.